cmake/tests.cmake

```diff
@@ -214,4 +214,5 @@ target_link_libraries(lite-arena-test libprotobuf-lite gmock_main)
 
 add_custom_target(check
   COMMAND tests
+  DEPENDS tests test_plugin
```

Comment:
I was wondering about that actually. Adding `tests` as a dependency did not seem to be empirically necessary to make the tests pass, but it seems like it ought to be there. If we leave it out, is CMake smart enough to determine that it should be a dependency since it appears as part of `COMMAND`?